File: LAB03 - Ticket management/server/src/ticketing/src/main/kotlin/com/lab3/ticketing/model/Ticket.kt
package com.lab3.ticketing.model



import com.lab3.ticketing.util.TicketState
import jakarta.persistence.*


@Entity
@Table
class Ticket(
    var state:TicketState,
    //@ManyToOne var customer:Profile,
    //@ManyToOne var expert:Profile,
    var description:String,
    //@ManyToOne var product:Product,
    @OneToMany var messages: MutableSet<Message>

): EntityBase<Long>() {

}

File: Final_Project/client/src/model/ticket.js
export class Ticket{
    constructor(id, state, customer, expert, product, messages, creationDate, lastModified) {
        this.id = id;
        this.state = state;
        this.customer = customer;
        this.expert = expert;
        this.product = product;
        this.messages = messages;
        this.creationDate = creationDate;
        this.lastModified = lastModified;
    }
}

